airgun/views/cloud_insights.py

```diff
@@ -71,3 +76,128 @@ class CloudInsightsView(BaseLoggedInView, SearchableViewMixinPF4):
+        """Return True if the select all checkbox is checked."""
+        return self.checkbox.selected
+
+
+class RemediateSummary(PF5OUIAModal):
+    """Models the Remediation summary page and button"""
+
+    title = PF5Title('Remediation summary')
+    remediate = PF5Button('Remediate')
+
+
+class RecommendationsDetailsView(BaseLoggedInView):
+    """Models everything in the recommendations details views execpt the affected system link"""
+
+    title = PF5Title('Affected Systems')
+    clear_button = PF5Button('Reset filters')
+    remediate = PF5Button('Remediate')
+    download_playbook = PF5Button('Download playbook')
+    search_field = TextInput(locator=('.//input[@aria-label="text input"]'))
+    bulk_select = BulkSelectMenuToggle()
+    table = PF5Table(
+        locator='.//table[contains(@aria-label, "Host inventory")]',
+        column_widgets={
+            0: Checkbox(locator='.//input[@type="checkbox"]'),
+            'Name': Text('.//a'),
+            'OS': Text('.//span'),
+            'Last seen': Text('.//span'),
+            'First impacted': Text('.//span'),
+        },
+    )
+
+    @property
+    def is_empty(self):
+        """Check whether the table is empty."""
+        return self.table.is_displayed
+
+    @property
+    def is_displayed(self):
+        return self.browser.wait_for_element(self.table, exception=False) is not None
+
+
+class RecommendationsTableExpandedRowView(RecommendationsDetailsView):
+    """View that models the recommendation expandable row content"""
+
+    affected_systems_url = Text(
+        locator='.//*[contains(@class, "ins-c-rule-details__view-affected")]//a'
+    )
+
+    @property
+    def is_displayed(self):
+        """Check that the row is displayed."""
+        return self.affected_systems_url.is_displayed
+
+
+class RecommendationsTabView(BaseLoggedInView):
+    """View representing the Recommendations Tab."""
+
+    title = PF5Title('Recommendations')
+    search_field = TextInput(locator=('.//input[@aria-label="text input"]'))
+    clear_button = PF5Button('Reset filters')
+    incidents = Text(locator='.//a[@data-testid="Incidents"]')
+    critical_recommendations = Text(locator='.//a[@data-testid="Critical recommendations"]')
+    important_recommendations = Text(locator='.//a[@data-testid="Important recommendations"]')
+    conditional_filter_dropdown = PF5OUIATextInput('ConditionalFilter')
+    table = PF5ExpandableTable(
+        locator='.//table[contains(@aria-label, "rule-table")]',
+        content_view=RecommendationsTableExpandedRowView,
+        column_widgets={
+            'Name': Text('.//a'),
+            'Modified': Text('.//span'),
+            'Category': Text('.//span'),
+            'Total risk': Text('.//span'),
+            'Systems': Text('.//div'),
+            'Remediation type': Text('.//span'),
+        },
+    )
+
+    @property
+    def is_empty(self):
+        """Check whether the table is empty."""
+        return self.table.is_displayed
```

Comment:
This doesn't actually check whether the table is empty or not. If we are going to include this, it should do what it says.